crates/l2/utils/eth_client/mod.rs

```diff
@@ -365,7 +365,9 @@ impl EthClient {
 
         // Add the nonce just if present, otherwise the RPC will use the latest nonce
         if let Some(nonce) = transaction.nonce {
-            data["nonce"] = json!(format!("{:#x}", nonce));
+            if let Value::Object(ref mut map) = data {
+                map.insert("nonce".to_owned(), json!(format!("{:#x}", nonce)));
```

Comment:
```suggestion
                map.insert("nonce".to_owned(), json!(format!("{nonce:#x}")));
```

Reply:
fixed in bec14139b55734f5f34037cefff5ff577c151829